src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -221,7 +221,7 @@ public ImmutableHashSet<T> Remove(T item)
         [Pure]
         public bool TryGetValue(T equalValue, out T actualValue)
         {
-            int hashCode = _equalityComparer.GetHashCode(equalValue);
+            int hashCode = _equalityComparer.GetHashCode(equalValue!);
```

Comment:
:memo: should we remove the `[DisallowNull]` on that `GetHashCode`, or add `[DisallowNull]` here?
(same question below)

Reply:
In general `IEqualityComparer<T>.GetHashCode` (e.g. StringComparer.GetHashCode throws ArgumentNullException for null) even if some implementations do, so we can't remove `[DisallowNull]`.  We could consider adding `[DisallowNull]` to this `T equalValue`, but then we'd be saying that null wasn't allowed even if the developer provided a comparer implementation that did actually allow null.  This is an example where we lack sufficient expressivity.  I guess for now it'd be best to add `[DisallowNull]`, and then see what kind of issues that causes; we can remove `[DisallowNull]` later on, but it'll be harder to add it later on.